### ISSUE TITLE
On windows10 Duckdb V0.2.9 will raise fatal error if we set threads > 1

### ISSUE THREAD

**Issue by thomasguo** [duckdb/duckdb#2344]
**What does happen?**
I imported the csv file into duckdb and executed some sql queries successfully. Then the program kept throwing a fatal error.  It seemed the database file has been corrupted for some reason. 

The python code is as follow:
```
conn = duckdb.connect('test.duckdb')
#conn.execute("COPY trips FROM 'D:/dataset/taxi/yellow_tripdata_2016-01.csv' ( HEADER )")
conn.execute("SELECT COUNT(VendorID) FROM trips")
```
After a few(about 10+) executions of the above code, the program raise an error:
RuntimeError: IO Error: Could not read all bytes from file "test.duckdb": wanted=262144 read=0

But the following sql query still works:
`
conn.execute("SELECT COUNT(*) FROM trips")
`

**What should happen?**
It should always execute query normally.

**To Reproduce**
Steps to reproduce the behavior. Bonus points if those are only SQL queries.
1.  download the infamous NY taxi trip dataset yellow_tripdata_2016-01.csv . 
2.  import the csv file into duckdb by local file mode.
3.  execute a simple aggregate sql for a few times.

**Environment (please complete the following information):**
 - OS: [windows 10]
 - DuckDB Version [0.2.9]
 - Python [3.8.10]

**Comment by Alex-Monahan** [comment]
I'm not certain, but you may need to close the connection in between each re-connect. Does using conn.close () help?

**Comment by thomasguo** [comment]
Thanks Alex.  Unfortunately conn.close() doesn't help.  Actually I did close the connection, which I did not show in above sample code.

**Comment by hannesmuehleisen** [comment]
What kind of environment is this running on? Is the file on a network drive?

**Comment by thomasguo** [comment]
@hannesmuehleisen
No, the file is just in local hard driver.   The test was ran at my old laptop with the following specs:

OS:  Windows 10 ,  revision 19042.1237
DuckDB Version:  0.2.9
Python: 3.8.10

Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz
RAM:  12GB DDL3
HDD 500GB(90GB spare)

**Comment by thomasguo** [comment]
more observation：
It seems OK with one month data (yellow_tripdata_2016-01.csv, 1.6GB).  The error usually occurs after I append the second month data (yellow_tripdata_2016-02.csv, 1.7GB).

**Comment by thomasguo** [comment]
More  testing and observation:
When the error occured,  I used  duckdb cli to open the database file:
`duckdb.exe test.duckdb` 
And I executed the exactly same sql statements there,  it worked! 

So I realized maybe the problem was on python side rather than on duckdb engine side, and the database file was not corrupted.

Then I reopen the dos prompt window and execute the python script again,  Surprisely this time it works again!

For now I  still don't understand what the exact root cause is. I will keep eyes on this case and maybe will change to another version python to test.  Any advices from you guys are welcome.

**Comment by thomasguo** [comment]
Am thinking I have located where the problem is. It's because I set threads parameter to boost the performance, as follow:

```
conn = duckdb.connect('test.duckdb')
#conn.execute("COPY trips FROM 'D:/dataset/taxi/yellow_tripdata_2016-01.csv' ( HEADER )")
conn.execute("PRAGMA memory_limit='1GB'")
#conn.execute("PRAGMA threads=4")
results = conn.execute("SELECT passenger_count, COUNT(*) FROM trips GROUP BY passenger_count")
print(results.fetchall())
conn.close()
```
The above code works well consistently.  If I uncomment the line of ”PRAGMA threads=4“  it throws that error accidentally. And once that occurs, it cannot recover until I comment that line again.   

Hope the information is helpful to figure out the root cause. 

PS1:  I have never seen that error occured in my linux enviroment.
PS2:  The PRAGMA threads statement does speed up some queries significantly.

**Comment by Mytherin** [comment]
I think I fixed this in #2691. Could you verify that that fixes the problem?

**Comment by thomasguo** [comment]
Thank you @Mytherin .   
I upgraded the duckdb module to latest version(0.3.2.dev330) via pip. Unfortunately the problem still exists. Error mesage is as follow:
RuntimeError: IO Error: IO Error: Could not read all bytes from file "test.duckdb": wanted=262144 read=0

**Comment by thomasguo** [comment]
Hi, @Mytherin, 

After further observation, I found that the problem occurs less frequently more regulary than before.  I'd like to describe my current test in steps as follow:
1)  I set the threads=4 and run the test several times, the problem always occurs.
2)  I set the threads=1 and run the test, the problem disappears.
3)  I set the threads=4 and run the test dozens of times, the problem never occurs.
4)  I run another monetdb-e test script. (this may force disk to clear cache)
5)  I run this duckdb test script(when the threads=4), the problem occurs again. The cycle goes back  to step 1.

From my point of view,  the problem seems to occur when it read file from disk. After that the file content exist in cache of disk(or something like that), the problem disappears.
